yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -62,10 +62,23 @@ private[spark] class ApplicationMaster(
     .asInstanceOf[YarnConfiguration]
   private val isClusterMode = args.userClass != null
 
-  // Default to numExecutors * 2, with minimum of 3
-  private val maxNumExecutorFailures = sparkConf.getInt("spark.yarn.max.executor.failures",
-    sparkConf.getInt("spark.yarn.max.worker.failures",
-      math.max(sparkConf.getInt("spark.executor.instances", 0) *  2, 3)))
+  // Default to numExecutors * 2 (maxExecutors in the case that we are
+  // dynamically allocating executors), with minimum of 3.
```

Comment:
this is hard to read. Can you rephrase:
```
// If dynamic allocation is enabled, default to the upper limit.
// Otherwise, default to 2 * numExecutors with a minimum of 3.
```